spring-cloud-stream-binder-test/src/main/java/org/springframework/cloud/stream/binder/AbstractBinderTests.java

```diff
@@ -190,9 +193,9 @@ public void handleMessage(Message<?> message) throws MessagingException {
 		moduleOutputChannel.send(message);
 		Assert.isTrue(latch.await(5, TimeUnit.SECONDS), "Failed to receive message");
 
-		assertThat(inboundMessageRef.get().getPayload()).isEqualTo("foo");
+		assertThat(new String(inboundMessageRef.get().getPayload(),StandardCharsets.UTF_8)).isEqualTo("foo");
```

Comment:
FYI, for future reference, AssertJ `isEqualTo` works with arrays so...

    assertThat(inboundMessageRef.get().getPayload()).isEqualTo("foo".getBytes());

...will work.

You don't really need a charset here since the payload is only ASCII characters.